sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala

```diff
@@ -139,12 +142,17 @@ object AkkaGrpcPlugin extends AutoPlugin {
       settings: Seq[String],
       generators: Seq[protocbridge.Generator]): Seq[protocbridge.Target] =
     generators.map { generator =>
-      protocbridge.Target(generator, targetPath, generator match {
-        case PB.gens.java => settings.filterNot(_ == "flat_package").filterNot(GeneratorOption.settings.contains)
-        case protocbridge.JvmGenerator("scala", ScalaPbCodeGenerator) =>
-          settings.filterNot(GeneratorOption.settings.contains)
-        case _ => settings
-      })
+      protocbridge.Target(
+        generator,
+        targetPath,
+        generator match {
+          case PB.gens.java =>
+            settings.filter(ProtocSettings.protocJava.contains)
```

Comment:
One thought: if we pass these indirectly via a sbt setting it would be possible to change it without a new release of the plugin (as a workaround)

Reply:
I like the idea, but in practice it requires passing around the values or wrapping functions in `Def.taskDyn` and friends - I think it's not worth it at this point.